Extensions/AzureTableAPI/Cosmos.DataTransfer.AzureTableAPIExtension/AzureTableAPIDataSinkExtension.cs

```diff
@@ -21,16 +21,72 @@ public async Task WriteAsync(IAsyncEnumerable<IDataItem> dataItems, IConfigurati
             var serviceClient = new TableServiceClient(settings.ConnectionString);
             var tableClient = serviceClient.GetTableClient(settings.Table);
 
-            await tableClient.CreateIfNotExistsAsync();
+            await tableClient.CreateIfNotExistsAsync(cancellationToken);
 
-            var createTasks = new List<Task>();
-            await foreach(var item in dataItems.WithCancellation(cancellationToken))
+            await Parallel.ForEachAsync(GetBatches(dataItems, settings), new ParallelOptions() { MaxDegreeOfParallelism = 8 }, async (batch, token) =>
             {
-               var entity = item.ToTableEntity(settings.PartitionKeyFieldName, settings.RowKeyFieldName);
-               createTasks.Add(tableClient.AddEntityAsync(entity));
+                await InnerWriteAsync(batch, tableClient, logger, token);
+            });
+        }
+
+        private static async IAsyncEnumerable<List<TableEntity>> GetBatches(IAsyncEnumerable<IDataItem> dataItems, AzureTableAPIDataSinkSettings settings)
+        {
+            var entities = new List<TableEntity>();
+            var first = true;
+            var partitionKey = string.Empty;
+
+            await foreach (var item in dataItems)
+            {
+                var tableEntity = item.ToTableEntity(settings.PartitionKeyFieldName, settings.RowKeyFieldName);
+
+                if (first)
+                {
+                    partitionKey = tableEntity.PartitionKey;
+                    first = false;
+                }
+
+                if (!tableEntity.PartitionKey.Equals(partitionKey) || entities.Count == 100)
+                {
+                    yield return entities;
+                    entities = new List<TableEntity>();
+                    partitionKey = tableEntity.PartitionKey;
+                }
+
+                entities.Add(tableEntity);
             }
 
-            await Task.WhenAll(createTasks);
+            if (entities.Count > 0)
+            {
+                yield return entities;
+            }
+        }
+
+        private static async Task InnerWriteAsync(List<TableEntity> tableEntities, TableClient tableClient, ILogger logger, CancellationToken cancellationToken)
+        {
+            var transactionsActions = tableEntities.Select(e => new TableTransactionAction(TableTransactionActionType.UpsertReplace, e));
```

Comment:
the only line that was changed was this one, TableTransactionActionType.Add became TableTransactionActionType.UpsertReplace